packages/account/src/Sections/Profile/PhoneVerification/resend-code-timer.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useRef } from 'react';
+import { Button, CaptionText } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+
+type TResendCodeTimer = {
+    resend_code_text: string;
+};
+const ResendCodeTimer = ({ resend_code_text }: TResendCodeTimer) => {
+    // TODO: Use dynamic value for setting initial value for counter once mockApi for timestamp is finalised
+    const [timer, setTimer] = React.useState(60);
+    const [start_timer, setStartTimer] = React.useState(false);
+    const request_code = useRef(resend_code_text);
+
+    React.useEffect(() => {
+        let countdown: NodeJS.Timeout;
+        if (start_timer && timer > 0) {
+            countdown = setInterval(() => {
+                setTimer(prevTime => prevTime - 1);
+                // TODO: change seconds to mins based on interval calculated using timestamp from mock api
+                request_code.current =
+                    resend_code_text === 'Resend code'
+                        ? `Resend code in ${timer - 1}s`
+                        : `Didn’t get the code?(${timer - 1}s)`;
+            }, 1000);
+        } else {
+            setStartTimer(false);
+            request_code.current = resend_code_text;
+        }
+
+        return () => clearInterval(countdown);
+    }, [timer, start_timer]);
+
+    const resendCode = () => {
+        setTimer(60);
+        setStartTimer(true);
+    };
+
+    return (
+        <Button variant='tertiary' onClick={resendCode} disabled={start_timer} color='black'>
+            <CaptionText bold underlined>
+                <Localize i18n_default_text='{{resendCode}}' values={{ resendCode: request_code.current }} />
```

Comment:
I'm not sure I understand, could you elaborate?

Reply:
@yauheni-deriv I think translation will work here. I remember last time I have similar scenario of this code and checked on the translation where it works fine 😎

Reply:
I think that values are considered dynamic values and they are not translated.
so for having this values translated, we need to pass them as translated.
so here in our case maybe it would be better just to refactor the code and avoid using Localize with values

Reply:
Please check with the rest of the team
@amina-deriv , @likhith-deriv , @suisin-deriv

Reply:
@yauheni-deriv you can check in poi-error-config, inside we have condition checks for `document_name` the translation works fine there i suppose this will work fine also

![Screenshot 2024-05-06 at 4 50 20 PM](https://github.com/suisin-deriv/deriv-app/assets/103026762/e35706f4-e509-4855-8e9f-0f91ba5cd126)

Reply:
I have my doubts on resend_code_text getting translated. As @yauheni-deriv said, the string has to be localized before passing it dynamically. Only values that do not need translation can be passed dynamically.

Reply:
would this work ? -> declaring a const which stores localised value of `request_code.current` and then pass inside <Localize>

Reply:
not sure I understand, but it is better to define separate variables with <Localize/>

btw, may I just confirm:
do we really need here ref? every second we will assign new value to it?

Reply:
Yes i think so, because we display a countdown that changes every second

Reply:
right, the context of my question is that ref is needed to store data between rerenders, and if the value in ref is changed, component is not rerendered. to rerender component we need to change state of component by calling setTimer, so we are doing it here. 
so there is not so much sense for ref here, because we change .current with every rerender. 
I think We can remove ref here and just render correct string conditionally and add timer value with setinerval and usestate.
how do you think?

Reply:
@yauheni-deriv @utkarsha-deriv I agree with @yauheni-deriv 's suggestion
We can refer this similar component "CountdownComponent" in  `packages/cfd/src/Containers/cfd-password-manager-modal.tsx`